packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts

```diff
@@ -173,7 +176,7 @@ export const verifyCredentialToAccept = async (args: VerifyCredentialToAcceptArg
 
   const verificationResult: VerificationResult = await verifyCredential(
     {
-      credential: credential as VerifiableCredential,
+      credential: wrappedVC.original as VeramoW3CVerifiableCredential, // FIXME figure out what type needs to go where
```

Comment:
Why did you make this change? `credential` comes from the actual VCI response, which in all cases will be the actual original credential

Reply:
because that type is not the correct one, we are using all sorts of casts which are not correct ones

Reply:
Okay but the original question stands. We have something incoming which is the original thing in the original code, then changing that to something which is constructed, needs a good reason. That you might not know how/which types to use at present might not be the best argument IMO. The problem simply stems from using multiple different APIs from different parties and extending these over time. As part of the OID4VC refactoring we hopefully start next quarter these things will be addressed.

I liked the original approach with an OriginalVerifiableCredential way better then now the VeramoW3CVerifiableCredential for instance